esphomeyaml/cookbook/dual-r2-cover.rst

```diff
@@ -0,0 +1,113 @@
+Dual relay cover motor control
+==============================
+
+The following is a possible configuration file for common covers that use a motor with 2 inputs. Only one should be powered at a time (interlocking) to either move the cover up or down. For this the [Sonoff Dual R2](https://www.itead.cc/sonoff-dual.html) can be used which has two independent relays. Additionally this configuration allows the single button on the Sonoff to control the motion by cycling between: open->stop->down->stop->...
+
+These kind of motors automatically stop when the end of the cover movement is reached. However, to be safe, this automation stops powering the motor after 1 minute of movement. In the rare case of the end-stop switch in the motor failing this will reduce the risk for damage or fire.
+
+
+.. code:: yaml
+
+  esphomeyaml:
+    name: cover
+    platform: espressif8266
+    board: esp01_1m
+    board_flash_mode: dout
+
+  wifi:
+    ssid: '***'
+    password: '***'
+
+  mqtt:
+    broker: 'mqtt'
+    username: ''
+    password: ''
+
+  logger:
+
+  ota:
+
+  binary_sensor:
+    - platform: gpio
+      pin: 10
+      name: "Cover button"
+      inverted: true
+      # 'hides' this control in homeassistant by not letting it be discovered
+      # since the cover can already be controller by the 'cover' component.
+      discovery: false
+      on_press:
+        then:
+          # logic for cycling through movements: open->stop->close->stop->...
+          - lambda: |
+              if (id(serre).state == cover::COVER_OPEN) {
+                if (id(open).value){
+                  id(serre).stop();
+                } else {
+                  id(serre).close();
+                }
+              } else if (id(serre).state == cover::COVER_CLOSED) {
+                if (id(close).value){
+                  id(serre).stop();
+                } else {
+                  id(serre).open();
+                }
+              } else {
+                if (id(open).value || id(close).value){
+                  id(serre).stop();
+                } else {
+                  id(serre).open();
+                }
+              }
+
+  switch:
+    - platform: gpio
+      pin: 12
+      name: open
+      id: open
+      # 'hide' this from homeassistant as we don't want it to be controller directly
+      discovery: false
+    - platform: gpio
+      pin: 5
+      name: close
+      id: close
+      # 'hide' this from homeassistant as we don't want it to be controller directly
+      discovery: false
```

Comment:
Hmmm, I probably should also add an option to make esphomelib not send *anything*. Currently this will cause esphomelib to not send discovery messages, but still send state messages.

Reply:
Agree, was going to raise a feature request for it. But as temporary workaround this was working enough for now :)